pennylane_lightning/lightning_base.py

```diff
@@ -16,35 +16,49 @@
 This module contains the base class for all PennyLane Lightning simulator devices,
 and interfaces with C++ for improved performance.
 """
+from pathlib import Path
 import numpy as np
+from pennylane.measurements import MeasurementProcess
 
+import pennylane_lightning
 from ._version import __version__
 
-try:
-    from .pennylane_lightning_ops import (
-        MeasurementsC64,
-        StateVectorC64,
-        MeasurementsC128,
-        StateVectorC128,
-        backend_info,
-    )
+libpath = Path(pennylane_lightning.__file__)
+libpath = libpath.parents[0].glob("pennylane_lightning_ops.*")
+CPP_BINARY_AVAILABLE = len(list(libpath)) > 0
 
-    CPP_BINARY_AVAILABLE = True
-except ModuleNotFoundError:
+if CPP_BINARY_AVAILABLE:
+    # pylint: disable=import-error, no-name-in-module, unused-import
+    from .pennylane_lightning_ops import backend_info
+else:
 
     def backend_info():
+        # pylint: disable=missing-function-docstring
         return {"NAME": "NONE"}
 
-    CPP_BINARY_AVAILABLE = False
+
+def _chunk_iterable(iteration, num_chunks):
+    "Lazy-evaluated chunking of given iterable from https://stackoverflow.com/a/22045226"
+    iteration = iter(iteration)
+    return iter(lambda: tuple(islice(iteration, num_chunks)), ())
+
 
 if CPP_BINARY_AVAILABLE:
     from typing import List
+    from itertools import islice, product
 
     import pennylane as qml
     from pennylane import QubitDevice
     from pennylane.operation import Operation
     from pennylane.wires import Wires
```

Comment:
Pylint doesn't complain about this?

Reply:
I guess not because `_chunk_iterable` is only called in subclasses when `CPP_BINARY_AVAILABLE == True`.